src/transaction/mod.rs

```diff
@@ -24,11 +26,43 @@ pub struct Timestamp {
     pub logical: i64,
 }
 
-pub enum Mutation {
-    Put(Key, Value),
-    Del(Key),
-    Lock(Key),
-    Rollback(Key),
+#[derive(Debug, Clone)]
+enum Mutation {
+    Put(Value),
+    Del,
+    Lock,
+}
+
+impl Mutation {
+    fn with_key(self, key: Key) -> kvrpcpb::Mutation {
+        let mut pb = kvrpcpb::Mutation {
+            key: key.into(),
+            ..Default::default()
+        };
+        match self {
+            Mutation::Put(v) => {
+                pb.set_op(kvrpcpb::Op::Put);
+                pb.set_value(v.into());
+            }
+            Mutation::Del => pb.set_op(kvrpcpb::Op::Del),
+            Mutation::Lock => pb.set_op(kvrpcpb::Op::Lock),
+        };
+        pb
+    }
+
+    /// Returns a `Some` if the value can be determined by this mutation. Otherwise, returns `None`.
```

Comment:
Maybe `Determined` and `Undetermined` here?

Reply:
Oh, I forgot to modify it.

Reply:
I think the type itself is clear enough so let's just remove this comment.